python/pyspark/sql/context.py

```diff
@@ -72,20 +91,25 @@ class SQLContext(object):
 
     _instantiatedContext = None
 
-    def __init__(self, sparkContext, sparkSession=None, jsqlContext=None):
+    def __init__(
+        self,
+        sparkContext: SparkContext,
+        sparkSession: Optional[SparkSession] = None,
+        jsqlContext: Optional[JavaObject] = None
+    ) -> None:
         if sparkSession is None:
             warnings.warn(
                 "Deprecated in 3.0.0. Use SparkSession.builder.getOrCreate() instead.",
                 FutureWarning
             )
 
         self._sc = sparkContext
-        self._jsc = self._sc._jsc
-        self._jvm = self._sc._jvm
+        self._jsc = self._sc._jsc  # type: ignore[attr-defined]
+        self._jvm = self._sc._jvm  # type: ignore[attr-defined]
```

Comment:
Should we add a note, that these could be removed, once `SparkContext` is inlined?